chore: Add in-package documentation (#127)
DataDog/dd-sdk-unity: packages/Datadog.Unity/Documentation/README.md

# Datadog Unity

The Datadog Unity SDK supports Real User Monitoring (RUM), logging, and crash reporting for Android and iOS apps built on Unity.

> Datadog does not support Desktop (Windows, Mac, or Linux), console, or web deployments from Unity. If you have a game or application and want to use Datadog RUM to monitor its performance, create a ticket with Datadog support.

Reviewer: ```suggestion
Datadog does not support desktop (Windows, Mac, or Linux), console, or web deployments from Unity. If you have a game or application and want to use Datadog RUM to monitor its performance, create a ticket with Datadog support.
```

## Setup

### Unity Setup

> To get started using the Datadog Unity SDK

Reviewer: ```suggestion
To get started using the Datadog Unity SDK:
```

> 1. [External Dependency Manager for Unity (EDM4U)](https://github.com/googlesamples/unity-jar-resolver). Datadog utilizes EDM4U to manage dependencies on Android and iOS.

Reviewer: ```suggestion
1. Install the [External Dependency Manager for Unity (EDM4U)](https://github.com/googlesamples/unity-jar-resolver). Datadog uses EDM4U to manage dependencies on Android and iOS.
```

> 2. Configure your project to use Gradle templates, and enable both Custom Main Template and Custom Gradle Properties Template.

Reviewer: ```suggestion
2. Configure your project to use Gradle templates, and enable both the Custom Main Template and Custom Gradle Properties Template.
```

> 3. If you build and receive Duplicate class errors (common in Unity 2022.x), add the following block in the dependencies block in your mainTemplate.gradle:

Reviewer: ```suggestion
3. If you build and receive any duplicate class errors (common in Unity 2022.x), add the following block in the dependencies block of your `mainTemplate.gradle` file:
```

```groovy
constraints {
     implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.8.0") {
         because("kotlin-stdlib-jdk8 is now a part of kotlin-stdlib")
     }
}
```

4. Ensure your Android application is using the [IL2CPP backend](https://docs.unity3d.com/6000.0/Documentation/Manual/scripting-backends-il2cpp.html) for Android builds.

### Create an application in Datadog

1. In Datadog, navigate to [Digital Experience > Add an Application](https://app.datadoghq.com/rum/application/create).
2. Choose Unity as the application type.
3. Provide an application name to generate a unique Datadog application ID and client token.
4. To disable automatic user data collection for either client IP or geolocation data, uncheck the boxes for those settings.

### Modify Datadog settings in Unity

After installing the Datadog Unity SDK, you need to set Datadog’s settings in the Unity UI. Navigate to your Project Settings and click on the Datadog section on the left hand side. You will see the following screen:

<img src="images/datadog-setup-ui.avif">

> The *Client Token* is required for any data to be sent to Datadog. The *RUM Application Id* is required to use any RUM features.

Reviewer: ```suggestion
- The *Client Token* is required for any data to be sent to Datadog.
- The *RUM Application Id* is required to use any RUM features.
```
